src/util/favicon.js

```diff
@@ -0,0 +1,89 @@
+import { DOMParser } from "@xmldom/xmldom";
+import isUrl from "is-url";
+
+/**
+ * Parse the HTML source into a DOM document
+ *
+ * @param {string | undefined} htmlSource the HTML source
+ * @returns {Document | undefined} the DOM document
+ */
+const parseHtmlSource = (htmlSource) => {
+  if (htmlSource && htmlSource.length > 0) {
+    // use a return statement for the error handler to avoid the console warning
+    // as any error will result in fallback favicon
+    return new DOMParser().parseFromString(htmlSource, "text/html");
+  }
+};
+
+/**
+ * Return the favicon URL from the HTML links collection
+ *
+ * @param {HTMLCollectionOf<Element> | undefined} links the HTML links collection
+ * @param {string} origin the origin URL to be used to build the favicon url
+ * @returns {string} the first found favicon URL or empty object if none found
+ */
+const getFaviconUrlFromLinks = (links, origin) => {
+  let faviconURL = "";
+
+  if (links && links.length > 0 && origin) {
+    for (let i in Array.from(links)) {
```

Comment:
You may use Array.every instead of a for here as it's far faster and would make it consistent with the app implem.
See https://github.com/MetaMask/metamask-mobile/pull/6951/files#diff-91801af11726fc69162ca9c2b97d3156f529d6581190773beba26b3e0fd4cb04R55-R63

Reply:
Hi @NicolasMassart,
I can argue that the `return false;` within the Array.every does not stop the loop at all, the Array.every will test all its elements against provided function.
https://github.com/MetaMask/metamask-mobile/pull/6951/files#diff-91801af11726fc69162ca9c2b97d3156f529d6581190773beba26b3e0fd4cb04R61
The rest of your comments has been sorted.

Reply:
I think returning false actually does stop every() and it's confirmed by [our unit tests](https://github.com/MetaMask/metamask-mobile/pull/6951/files#diff-24d9b3e8d6ae7de0e04f40e6b451f58ab6628a79549cc884f51ad22a75db7205R137-R152).
> The every() method is an [iterative method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array#iterative_methods). It calls a provided callbackFn function once for each element in an array, until the callbackFn returns a [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) value. If such an element is found, every() immediately returns false and stops iterating through the array.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every#description

Can you elaborate if you really think this is wrong and I misunderstood the behaviour please?

Reply:
My apologies, I must have missed that part, the code has been updated.